jgrapht-core/src/main/java/org/jgrapht/util/SupplierUtil.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * (C) Copyright 2018-2018, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.util;
+
+import java.io.Serializable;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DefaultWeightedEdge;
+
+/**
+ * Helper class for suppliers.
+ *
+ * @author Dimitrios Michail
+ */
+public class SupplierUtil
+{
+
+    /**
+     * Supplier for {@link DefaultEdge}.
+     */
+    public static final Supplier<DefaultEdge> DEFAULT_EDGE_SUPPLIER =
```

Comment:
Analogous to the other methods, wouldn't it be cleaner to have `createDefaultEdgeSupplier`, `createDefaultWeightedEdgeSupplier`, etc methods?
That way, we would have:
`Graph<Integer, DefaultEdge> grid = new SimpleGraph<>(SupplierUtil.createIntegerSupplier(), SupplierUtil.createDefaultEdgeSupplier(), false);
instead of:
`Graph<Integer, DefaultEdge> grid = new SimpleGraph<>(SupplierUtil.createIntegerSupplier(), SupplierUtil.DEFAULT_EDGE_SUPPLIER, false);`

Reply:
Also added methods `SupplierUtil.createDefaultEdgeSupplier()` which create a new supplier each time.